go/array_string/compression/compress.go

```diff
@@ -0,0 +1,42 @@
+// Package compress contains implementation of compression algorithms
+package compress
+
+import (
+	"strconv"
+	"strings"
+)
+
+// String - compresses the input and return it
+func String(input string) string {
+	// use string builder to avoid memory allocation
+	// required to join new and old string
+	var output strings.Builder
+	charCount := make(map[string]int)
```

Comment:
You don't need a whole map for this because you are always deleting it. Just use an `int` and reset it to 0 when the next character doesn't match.

Reply:
That's true, thank you!